src/Symfony/Component/DependencyInjection/Compiler/AttributeAutoconfigurationPass.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ChildDefinition;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * @author Alexander M. Turek <me@derrabus.de>
+ */
+final class AttributeAutoconfigurationPass implements CompilerPassInterface
+{
+    private $ignoreAttributesTag;
+
+    public function __construct(string $ignoreAttributesTag = 'container.ignore_attributes')
+    {
+        $this->ignoreAttributesTag = $ignoreAttributesTag;
+    }
+
+    public function process(ContainerBuilder $container): void
+    {
+        if (80000 > \PHP_VERSION_ID) {
+            return;
+        }
+
+        $autoconfiguredAttributes = $container->getAutoconfiguredAttributes();
+
+        foreach ($container->getDefinitions() as $id => $definition) {
+            if (!$definition->isAutoconfigured()
+                || $definition->isAbstract()
+                || $definition->hasTag($this->ignoreAttributesTag)
+                || !($reflector = $container->getReflectionClass($definition->getClass(), false))
+            ) {
+                continue;
+            }
+
+            $instanceof = $definition->getInstanceofConditionals();
+            $conditionals = $instanceof[$reflector->getName()] ?? new ChildDefinition('');
+            foreach ($reflector->getAttributes() as $attribute) {
+                if (!($configurator = $autoconfiguredAttributes[$attribute->getName()] ?? null)) {
```

Comment:
we don't need the brackets just after the `!`, do we?

Reply:
or we might remove the `!` and put the logic inside this block btw - I'm not sure a guard statement brings enough here

Reply:
In an earlier iteration, the `foreach` body used to be larger. I agree, we can invert the logic now.